src/controller/CHIPDeviceController.cpp

```diff
@@ -931,6 +946,59 @@ void DeviceCommissioner::OnSessionEstablished()
     RendezvousCleanup(CHIP_NO_ERROR);
 }
 
+CHIP_ERROR DeviceCommissioner::SendOperationalCertificateSigningRequestCommand(NodeId remoteDeviceId)
+{
+    // TODO: Call OperationalCredentials cluster API to send command
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DeviceCommissioner::OnOperationalCertificateSigningRequest(NodeId node, const ByteSpan & csr)
+{
+    ReturnErrorCodeIf(mOperationalCredentialsDelegate == nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    auto opCertBuf     = static_cast<uint8_t *>(chip::Platform::MemoryAlloc(kMaxCHIPOpCertLength));
+    auto opCert        = std::unique_ptr<uint8_t>(opCertBuf);
```

Comment:
This has to be unique_ptr<uint8_t[]> to call delete[] instead of just delete.
same for below.

You also need to provide a deleter - this works fine apparently on malloc-based platform allocators but will fail on all others as std::unique_ptr will do delete instead of  Platform::MemoryFree

Reply:
Updated the code to use ScopedBuffer.